File: 240520/피보나치 수/fibonacci-number.java
import java.util.*;

public class Main {
    static Scanner sc = new Scanner(System.in);
    static int n = sc.nextInt();
    static int[] tabDp = new int[n + 1]; 

    public static void main(String[] args) {
        if(n == 1) {
            System.out.println(1);
            return;
        }

        tabDp[1] = 1;
        for(int i = 2; i <= n; i++) {
            tabDp[i] = tabDp[i-2] + tabDp[i-1];
        }
        System.out.println(tabDp[n]);
    }
}

File: 240520/피보나치 수/fibonacci-number.kt
import java.util.*

val sc = Scanner(System.`in`)
val n = sc.nextInt()
val tabDp = IntArray(n + 1) {0}

fun main() {
    tabDp[1] = 1
    if(n == 1) {
        println(1)
        return
    }

    for(i in 2 .. n) {
        tabDp[i] = tabDp[i-2] + tabDp[i-1]
    }
    println(tabDp[n])
}